presto-main/src/main/java/com/facebook/presto/cost/AssignUniqueIdStatsRule.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cost;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.matching.Pattern;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.iterative.Lookup;
+import com.facebook.presto.sql.planner.plan.AssignUniqueId;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+
+public class AssignUniqueIdStatsRule
+        implements ComposableStatsCalculator.Rule
```

Comment:
@sopel39 Why `ComposableStatsCalculator.Rule` is not a generic with `PlanNode` parameter?

```
public class AssignUniqueIdStatsRule 
                 implements ComposableStatsCalculator.Rule<AssignUniqueId>`
```

would be better. Then `(AssignUniqueId) node` cast in `apply` won't be needed.

Reply:
It could be, just like in `com.facebook.presto.sql.planner.iterative.Rule`. @mbasmanova do you want to fix this?

Reply:
@findepi There are not too many places to update yet. Here you go: #10748